src/array.vala

```diff
@@ -41,11 +41,8 @@ public class Vast.Array : Object
     /* GObject that owns the memory buffer for storage of scalar elements */
     public Bytes? data {get; construct; }
 
-    /* global shift from the _cached_data; _cached_data + origin is the memory location of 0-th element */
-    public size_t origin { get; construct; default = 0; }
-
-    /* pointer to the begining memory buffer, */
-    internal unowned uint8* _cached_data;
+    /* pointer to the memory location of 0th element */
+    public uint8 * origin {get; construct; default = null;}
```

Comment:
It seems to me that the day we'll extract an abstract class, we'll have to have specialized `Iterator` implementation as well to take advantage of the specific bits of each implementation.